### ISSUE TITLE
Add `namespace` method to `NamespacedCorestore`

### ISSUE THREAD

**Issue by RangerMauve** [holepunchto/corestore#10]
Would a PR that adds the `namespace` method to `NamespacedCorestore` be welcome?

The reasoning is that it'd be useful for NamespacedCorestore to have the same public interface as a regular corestore, e.g. if a data structure that uses namespacing wants more namespaced stores, it should be able to do so without needing a reference to the non-namespaced store.

Regarding behaviour, would it be better to have it invoke `this.store.namespace(namespace` directly, or would it be a good idea to prefix with the current namespace like `this.store.namespace(this.name + namespace)`?

**Comment by andrewosh** [comment]
Hey @RangerMauve sorry I missed this one. Yeah it'd be good to have this just to keep things intuitive! PR would be more than welcome, otherwise I'd be happy to throw it together

**Comment by Frando** [comment]
Nice, I was just going to file exactly this issue :-)

In [kappa-record-db](https://github.com/arso-project/kappa-record-db/blob/master/group.js#L72) I currently have these lines:
```javascript
    // Patch in a recursive namespace method if we got a namespaced corestore.
    if (!this.corestore.namespace && this.corestore.store) {
      this.corestore.namespace = name => this.corestore.store.namespace(this.corestore.name + ':' + name)
    }
```
so +1 to adding this

**Comment by RangerMauve** [comment]
I don't think I'll have badwidth to do a PR for this any time soon. 😅

**Comment by Frando** [comment]
--> #13

**Comment by andrewosh** [comment]
Released in 5.2.0. Thanks for the impl @Frando
